tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb

```diff
@@ -121,111 +120,120 @@ def run(opts)
+                software_name = vuln.fetch("softwareName") if vuln.key?("softwareName")
+                software_version = vuln.dig(:softwareVersion)
+                vulnerability_severity_level = vuln.fetch("vulnerabilitySeverityLevel") if vuln.key?("vulnerabilitySeverityLevel")
+                recommended_security_update = vuln.fetch("recommendedSecurityUpdate") if vuln.key?("recommendedSecurityUpdate")
+                recommended_security_update_id = vuln.fetch("recommendedSecurityUpdateId") if vuln.key?("recommendedSecurityUpdateId")
+                recommended_security_update_url = vuln.fetch("recommendedSecurityUpdateUrl") if vuln.key?("recommendedSecurityUpdateUrl")
+                disk_paths = vuln.fetch("diskPaths") if vuln.key?("diskPaths")
+                registry_paths = vuln.fetch("registryPaths") if vuln.key?("registryPaths")
+                end_of_support_status = vuln.fetch("endOfSupportStatus") if vuln.key?("endOfSupportStatus")
+                end_of_support_date = vuln.fetch("endOfSupportDate") if vuln.key?("endOfSupportDate")
+                exploitability_level = vuln.fetch("exploitabilityLevel") if vuln.key?("exploitabilityLevel")
+                recommendation_reference = vuln.fetch("recommendationReference") if vuln.key?("recommendationReference")
+
+                details = {
+                  "softwareVendor" => software_vendor,
+                  "softwareName" => software_name,
+                  "softwareVersion" => software_version,
+                  "vulnerabilitySeverityLevel" => vulnerability_severity_level,
+                  "recommendedSecurityUpdate" => recommended_security_update,
+                  "recommendedSecurityUpdateId" => recommended_security_update_id,
+                  "recommendedSecurityUpdateUrl" => recommended_security_update_url,
+                  "diskPaths" => disk_paths,
+                  "registryPaths" => registry_paths,
+                  "endOfSupportStatus" => end_of_support_status,
+                  "endOfSupportDate" => end_of_support_date,
+                  "exploitabilityLevel" => exploitability_level,
+                  "recommendationReference" => recommendation_reference
+                }
+
+                details.compact!
+
+                vuln_object = {
+                  "scanner_identifier" => scanner_id,
+                  "scanner_type" => "MS Defender TVM",
+                  # scanner score should fallback using criticality (in case of missing cvss)
+                  "scanner_score" => vuln_score,
+                  "details" => JSON.pretty_generate(details)
+                }
+
+                # craft the vuln def hash
+                vuln_def = {
+                  "scanner_identifier" => scanner_id,
+                  "scanner_type" => "MS Defender TVM",
+                  "name" => vuln_name
+                }
+                vuln_def[:cve_identifiers] = vuln_cve.to_s if !vuln_cve.nil? && !vuln_cve.empty?
```

Comment:
I would recommend using this instead
`vuln_def[:cve_identifiers] = vuln_cve.to_s unless vuln_cve.nil? && vuln_cve.empty?`

The result is the same, but we got rid of two objections.

Reply:
I didn't touch this code but let me know if would still be needed?